packages/electric-db-collection/src/electric.ts

```diff
@@ -80,6 +92,22 @@ function isMustRefetchMessage<T extends Row<unknown>>(
   return isControlMessage(message) && message.headers.control === `must-refetch`
 }
 
+function isSnapshotEndMessage<T extends Row<unknown>>(
+  message: Message<T>
+): message is ControlMessage & { headers: { control: `snapshot-end` } } {
+  return isControlMessage(message) && message.headers.control === `snapshot-end`
+}
+
+function parseSnapshotMessage(
+  message: ControlMessage & { headers: { control: `snapshot-end` } }
```

Comment:
nit: perhaps worth introducing a helper type `type SnapshotEndMessage = ControlMessage & { headers: { control: `snapshot-end` } }` since we used this type twice already